api/desecapi/views.py

```diff
@@ -120,6 +122,95 @@ class DomainDetailByName(DomainDetail):
     lookup_field = 'name'
 
 
+class RRsetDetail(generics.RetrieveUpdateDestroyAPIView):
+    lookup_field = 'subname'
+    serializer_class = RRsetSerializer
+    permission_classes = (permissions.IsAuthenticated, IsDomainOwner,)
+
+    def get_queryset(self):
+        name = self.kwargs['name']
+        subname = self.kwargs['subname']
+        type = self.kwargs['type']
+
+        return RRset.objects.filter(domain__name=name, subname=subname, type=type)
+
+    def delete(self, request, *args, **kwargs):
+        try:
+            return super().delete(request, *args, **kwargs)
+        except django.core.exceptions.ValidationError as e:
+            ex = ValidationError(detail={"detail": str(e)})
+            ex.status_code = status.HTTP_409_CONFLICT
+            raise ex
+
+    def get(self, request, *args, **kwargs):
+        subname = self.kwargs['subname']
+        type = self.kwargs['type'].upper()
+
+        # TODO Do we need this functionality at all? We could just disregard the SOA record completely.
+        if type == 'SOA' and subname == '':
```

Comment:
aggregate those SOA checks in one place, currently they are split across different layers